From 368bb886ad1185dab40008ed7a05db65f589353d Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Thu, 6 Jun 2024 22:11:45 +0800
Subject: [PATCH 097/149] Revert "AP MLD: Add MLO Link KDE for each affiliated
 link in EAPOL-Key 3/4"

This reverts commit df59880042cd8d9b4bdd2dce6de0a6e233be1b64.

Please noted that this commit is a workaround for MTK STA IoT issue within WiFi7 R1 certification.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>

---
 src/ap/wpa_auth.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/ap/wpa_auth.c b/src/ap/wpa_auth.c
index 72a54844c..4b6d55a9e 100644
--- a/src/ap/wpa_auth.c
+++ b/src/ap/wpa_auth.c
@@ -4486,6 +4486,12 @@ static size_t wpa_auth_ml_kdes_len(struct wpa_state_machine *sm)
 		struct wpa_authenticator *wpa_auth;
 		const u8 *ie;
 
+		/* FIXME: This is a temporary workaround for MTK
+		 * sta IoT issue in WiFi7 cert.
+		 */
+		if (!sm->mld_links[link_id].valid)
+			continue;
+
 		wpa_auth = wpa_get_link_auth(sm->wpa_auth, link_id);
 		if (!wpa_auth)
 			continue;
@@ -4551,6 +4557,12 @@ static u8 * wpa_auth_ml_kdes(struct wpa_state_machine *sm, u8 *pos)
 		size_t rsne_len, rsnxe_len, rsnoe_len, rsno2e_len, rsnxoe_len;
 		size_t kde_len;
 
+		/* FIXME: This is a temporary workaround for MTK
+		 * sta IoT issue in WiFi7 cert.
+		 */
+		if (!sm->mld_links[link_id].valid)
+			continue;
+
 		wpa_auth = wpa_get_link_auth(sm->wpa_auth, link_id);
 		if (!wpa_auth)
 			continue;
-- 
2.45.2

